src/ape/managers/networks.py

```diff
@@ -92,15 +97,19 @@ def fork(
         Returns:
             :class:`~ape.api.networks.ProviderContextManager`
         """
-        forked_network = self.ecosystem.get_network(f"{self.network.name}-fork")
-        provider_settings = provider_settings or {}
+        try:
+            forked_network = self.ecosystem.get_network(f"{self.network.name}-fork")
+        except NetworkNotFoundError as err:
+            raise NetworkError(f"Unable to fork network '{self.network.name}'.") from err
```

Comment:
though: Cool, I wasn't aware of the `raise ... from` syntax.

Reply:
Ya it preserves the exception stack when raising a new error in an except block, it is nifty